src/sharding/ShardClientUtil.js

```diff
@@ -224,6 +226,18 @@ class ShardClientUtil {
     }
     return this._singleton;
   }
+
+  /**
+   * Get the shard Id for a given guild Id.
+   * @param {Snowflake} guildId Snowflake guild Id to get shard Id for
+   * @param {number} shardCount Number of shards
+   * @returns {number}
+   */
+  static shardIdForGuildId(guildId, shardCount) {
+    const shard = Number(BigInt(guildId) >> 22n) % shardCount;
+    if (shard < 0 || shard >= shardCount) throw new Error('SHARDING_SHARD_MISCALCULATION', guildId, shardCount, shard);
```

Comment:
Since `shard` is calculated with the modulus of `shardCount`, it can never be higher or equals than it, therefore only negative numbers would trigger this condition:

```suggestion
    if (shard < 0) throw new Error('SHARDING_SHARD_MISCALCULATION', guildId, shardCount, shard);
```

Reply:
Ah, yes